src/Queue/BunnyProducer.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Skrz\Bundle\BunnyBundle\Queue;
+
+use Bunny\Channel;
+use ReflectionClass;
+use Skrz\Bundle\BunnyBundle\Exception\BunnyException;
+use Skrz\Bundle\BunnyBundle\Service\BunnyManager;
+use Skrz\Bundle\BunnyBundle\Service\ContentTypes;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * @author Lukas Senfeld <skrz@senfeld.net>
+ * @method preProcessMessage(mixed $message, Channel $channel): void
+ */
+abstract class BunnyProducer implements BunnyProducerInterface
+{
+
+	private const NAME_SUFFIX_TO_REMOVE = "Producer";
+	private BunnyManager $manager;
+	private SerializerInterface $serializer;
+	private bool $autoCommit = false;
+	private string $exchange = "";
+	private bool $immediate = false;
+	private bool $mandatory = false;
+	private string $contentType = ContentTypes::APPLICATION_JSON;
+	private string $messageClassName;
+	private string $name;
+	private string $routingKey = "";
+	private bool $transactional = false;
+	private bool $hasBeforeMethod;
+
+	public function __construct(?string $name = null)
+	{
+		$this->name = $name ?? $this->generateName();
+		$this->hasBeforeMethod = method_exists($this, 'preProcessMessage');
+		$this->configure();
+	}
+
+	public function inject(BunnyManager $manager, SerializerInterface $serializer): void
```

Comment:
Proč se tyhle závislosti vkládají přes `inject` a ne konstruktorem?

Reply:
pokud bych je tady dal construktorem tak vsechny COnsumery v aplikacich by to museli reflektovat cemuz jsem se chtel vyhnout takhle kdyz udelas `MyCustomProducer extends BunnyProducer` tak muzes pouzit construktor dependenci aniz by jsi musel reflektovat predka. Je to hlavne kvuli tomu ze kdyby se ukazalo ze tady budu potrebovat dalsi zavyslost jako treba `EventDispatcher` tak neudelam BC v konstruktoru